File: 프로그래머스/lv2/42746. 가장 큰 수/가장 큰 수.js
function solution(numbers) {
    // 첫 번째 풀이
    // numbers = numbers.map( el => el.toString() );
    // numbers.sort((a, b) => Number(b + a) - Number(a + b));
    // return numbers[0] !== '0' ? numbers.join('') : '0';
    // 다른 분 풀이
    const buckets = [];

    for (const n of numbers){
        const s = String(n), l = s.length;

        let j = '';
        for (let i = 0; i < 4; i++){
            j += s[l > i ? i : i % l];
        }
        j = 9999 - j;

        buckets[j] = buckets[j] ? buckets[j] + s : s;
    }

    const ans = buckets.join``;

    return ans[0] === '0' ? '0' : ans;

}

	

File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.java
import java.util.*;

class Solution {
    public String solution(int[] numbers) {
        Integer[] numArray = new Integer[numbers.length];
        for (int i = 0; i < numbers.length; i++) {
            numArray[i] = numbers[i];
        }

        Arrays.sort(numArray, (a, b) -> {
            int ab = Integer.parseInt("" + a + b);
            int ba = Integer.parseInt("" + b + a);
            return Integer.compare(ba, ab);
        });

        if (numArray[0] == 0) {
            return "0";
        }

        StringBuilder answer = new StringBuilder();
        for (int num : numArray) {
            answer.append(num);
        }

        return answer.toString();
    }
}